e2e/common/createCitizenUser/idamApiHelper.ts

```diff
@@ -0,0 +1,106 @@
+import { APIRequestContext, request } from "@playwright/test";
+import * as dotenv from "dotenv";
+import qs from "qs";
+import { v4 as uuidv4 } from "uuid";
+
+dotenv.config();
+
+/**
+ * Initializes the Playwright API request context.
+ * @returns {Promise<APIRequestContext>}
+ */
+export async function initializeAPIContext(): Promise<APIRequestContext> {
+  return await request.newContext();
+}
+
+/**
+ * Function to get an access token from the IDAM service
+ * @param {APIRequestContext} apiContext The API request context
+ * @returns {Promise<string | null>} The access token if successful, otherwise null
+ */
+export async function getAccessToken(
+  apiContext: APIRequestContext,
+): Promise<string | null> {
+  try {
+    const data = {
+      grant_type: "client_credentials",
+      client_id: "prl-cos-api",
+      client_secret: process.env.IDAM_SECRET,
+      scope: "profile roles",
+    };
+    const response = await apiContext.post(
+      process.env.IDAM_TOKEN_URL as string,
+      {
+        headers: { "content-type": "application/x-www-form-urlencoded" },
+        data: qs.stringify(data),
+      },
+    );
+    if (response.ok()) {
+      const responseData = await response.json();
+      return responseData.access_token;
+    } else {
+      console.error(
+        "Error fetching access token:",
+        response.status(),
+        await response.text(),
+      );
+      return null;
+    }
+  } catch (error) {
+    console.error("Error fetching access token:", error);
+    return null;
+  }
+}
+
+/**
+ * Function to create a citizen user
+ * @param {APIRequestContext} apiContext The API request context
+ * @param {string} token The access token
+ * @returns {Promise<{ email: string; password: string; id: string }>} The created user's details
+ */
+export async function createCitizenUser(
+  apiContext: APIRequestContext,
+  token: string,
+): Promise<{ email: string; password: string; id: string }> {
+  if (!process.env.IDAM_CITIZEN_USER_PASSWORD) {
+    throw new Error("PASSWORD environment variable is not defined");
+  }
+  const uniqueId = uuidv4();
+  const id = uniqueId;
+  const password = process.env.IDAM_CITIZEN_USER_PASSWORD as string;
+  const email = `TEST_PRL_USER_citizen-user.${uniqueId}@test.local`;
+
+  const response = await apiContext.post(
+    process.env.IDAM_TESTING_SUPPORT_USERS_URL as string,
+    {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+      data: {
+        password,
+        user: {
+          id,
+          email,
+          forename: "fn_" + uniqueId.split("-")[0],
+          surname: "sn_" + uniqueId.split("-")[1],
+          roleNames: ["citizen"],
+        },
+      },
+    },
+  );
+  if (response.ok()) {
+    const responseData = await response.json();
+    console.log("User created:", responseData);
```

Comment:
Do we need to be logging this? It's just going to mess up our logs for debugging actual failures

Reply:
was thinking it would be useful to know the log in details in case a test fails and you want to log in to debug